Common/VR/VRBase.cpp

```diff
@@ -5,14 +5,103 @@
 #include <string.h>
 #include <unistd.h>
 
+#ifdef OPENXR_PLATFORM_PICO
+#define XR_PICO_ANDROID_CONTROLLER_FUNCTION_EXT_ENABLE_EXTENSION_NAME "XR_PICO_android_controller_function_ext_enable"
+#define XR_PICO_VIEW_STATE_EXT_ENABLE_EXTENSION_NAME "XR_PICO_view_state_ext_enable"
+#define XR_PICO_FRAME_END_INFO_EXT_EXTENSION_NAME "XR_PICO_frame_end_info_ext"
+#define XR_PICO_CONFIGS_EXT_EXTENSION_NAME "XR_PICO_configs_ext"
+#define XR_PICO_RESET_SENSOR_EXTENSION_NAME "XR_PICO_reset_sensor"
+
+enum ConfigsSetEXT
+{
+    UNREAL_VERSION = 0,
+    TRACKING_ORIGIN,
+    OPENGL_NOERROR,
+    ENABLE_SIX_DOF,
+    PRESENTATION_FLAG,
+    ENABLE_CPT,
+    PLATFORM,
+    FOVEATION_LEVEL,
+    SET_DISPLAY_RATE = 8,
+    MRC_TEXTURE_ID = 9,
+};
+
+typedef enum
+{
+    PXR_HMD_3DOF = 0,
+    PXR_HMD_6DOF
+} PxrHmdDof;
+
+typedef enum
+{
+    PXR_CONTROLLER_3DOF = 0,
+    PXR_CONTROLLER_6DOF
+} PxrControllerDof;
+
+typedef XrResult (XRAPI_PTR *PFN_xrSetConfigPICO) (
+XrSession                             session,
+enum ConfigsSetEXT                    configIndex,
+char *                                configData);
+PFN_xrSetConfigPICO    pfnXrSetConfigPICO;
+
+//cmc ext function ,not use from 2021/07
+typedef XrResult (XRAPI_PTR *PFN_xrSetEngineVersionPico)(XrInstance instance,const char* version);
+typedef XrResult (XRAPI_PTR *PFN_xrStartCVControllerThreadPico)(XrInstance instance,int headSensorState, int handSensorState);
+typedef XrResult (XRAPI_PTR *PFN_xrStopCVControllerThreadPico)(XrInstance instance,int headSensorState, int handSensorState);
+
+XrInstance mControllerInstance;
+PFN_xrSetEngineVersionPico pfnXrSetEngineVersionPico = NULL;
+PFN_xrStartCVControllerThreadPico pfnXrStartCVControllerThreadPico = NULL;
+PFN_xrStopCVControllerThreadPico pfnXrStopCVControllerThreadPico = NULL;
+
+inline void InitializeGraphicDeivce(XrInstance mInstance) {
+    mControllerInstance = mInstance;
+    xrGetInstanceProcAddr(mInstance, "xrSetEngineVersionPico", (PFN_xrVoidFunction*)(&pfnXrSetEngineVersionPico));
+    xrGetInstanceProcAddr(mInstance, "xrStartCVControllerThreadPico", (PFN_xrVoidFunction*)(&pfnXrStartCVControllerThreadPico));
+    xrGetInstanceProcAddr(mInstance, "xrStopCVControllerThreadPico", (PFN_xrVoidFunction*)(&pfnXrStopCVControllerThreadPico));
+}
+
+inline int Pxr_SetEngineVersion(const char *version) {
+    if (pfnXrSetEngineVersionPico != NULL) {
+        return pfnXrSetEngineVersionPico(mControllerInstance,version);
+    } else {
```

Comment:
minor nit: I would generally recommend this pattern:

```c++
int foo() {
	if (bar != nullptr) {
		return 42;
	}
	return 1337;
}
```

Over an else, mostly to reduce indentation.  Also nullptr rather than old-style NULL.

-[Unknown]

Reply:
I took this from Pico SDK sample but I will modify it to use PPSSPP code standards.